server/datastore/mysql/vpp.go

```diff
@@ -495,8 +500,8 @@ ON DUPLICATE KEY UPDATE
 	if insertOnDuplicateDidInsertOrUpdate(res) {
 		id, _ = res.LastInsertId()
 	} else {
-		stmt := `SELECT id FROM vpp_apps_teams WHERE adam_id = ? AND platform = ?`
-		if err := sqlx.GetContext(ctx, tx, &id, stmt, appID.AdamID, appID.Platform); err != nil {
+		stmt := `SELECT id FROM vpp_apps_teams WHERE adam_id = ? AND platform = ? AND global_or_team_id = ?`
```

Comment:
Yay for finding another unreleased bug while fixing the first one 😅

Reply:
this is what had me spinning for so long!! I happened to have the same VPP app added to a team _and_ to no team, which is what revealed this issue. I could not figure out for the life of me what was going on but finally narrowed it down to this 🙏